### ISSUE TITLE
Custom requires

### ISSUE THREAD

**Issue by mottosso** [mottosso/rez-pipz#14]
### Goal

Bridge the gap between pip and Rez "requirements"

### Motivation

I've got a Python library which depends on Git, but Git is not a Python package. Only Rez can make such requirements.

### Implementation

```bash
# Overwrite
rez env pipz -- install . --requires=git
rez env pipz -- install . --requires git mongo plasma

# More..
rez env pipz -- install . --add-requires=git
rez env pipz -- install . --remove-requires=six
```

**Comment by bpabel** [comment]
May be something that could benefit from using the new `pyproject.toml` file so that the project can define the dependency instead of the end user having to know.

**Comment by mottosso** [comment]
Yes, as I was experimenting with this, the first problem was specifying different requirements per version being installed, as one forgets or never knew it was supposed to have additional/different ones.

In what way can do you see pyproject.toml helping here? Is it able to define third-party requirements of sorts? One of the issues with relying on the project itself is that it would only apply to projects that have taken Rez into account; so our own projects work fine, but then public projects are "limited" in specifying Python requirements only. Some examples being projects like pygit or pymongo.

**Comment by bpabel** [comment]
`pyproject.toml` is basically just a standardized generic config and metadata file for python packages.  Right now, the main use case is for defining build commands, dependencies, and options, but you can put anything in there and give it your own section.

[`poetry`](https://poetry.eustace.io/docs/pyproject/), [`flit`](https://poetry.eustace.io/docs/pyproject/) , and the python [`black`](https://github.com/python/black#pyprojecttoml) tools all read settings from your project's `pyproject.toml` file.  So you could define your own section to pull settings from:

```
[tool.pipz]
requires = ["git", "mongo", "plasma"]
```
Or something like that.

**Comment by mottosso** [comment]
Have thought about this, and that's probably the better way to go. Hopefully one can get `setup.py` to provide similar functionality, for those stuck in the 00s. Even if that means text-parsing it retroactively. Thanks @bpabel.
